op-batcher/batcher/service.go

```
@@ -97,20 +97,15 @@ type BatcherService struct {
 
 type DriverSetupOption func(setup *DriverSetup)
 
-const (
-	attestationCID  = 3    // Well-known vsock CID for the attestation agent
-	attestationPort = 8000 // Default port for the attestation agent
-)
-
 func isRunningInEnclave() bool {
-	// Attempt to dial the Nitro Enclaves attestation agent via vsock.
-	// We use a short timeout to avoid blocking too long if not present.
-	conn, err := vsock.Dial(attestationCID, attestationPort, &vsock.Config{})
+	// Check whether `/sys/devices/virtual/misc/nitro_enclaves/` exists.
+	// If the directory does not exist, you're inside an enclave.
+	// If it does exist, you're outside the enclave.
+	fi, err := os.Stat("/sys/devices/virtual/misc/nitro_enclaves/")
 	if err != nil {
-		return false
+		return true
 	}
```

Does this case mean we aren't able to tell whether `fi` is a directory? Shouldn't we return an error or `false` in this case?

---

[da01c1a](https://github.com/EspressoSystems/optimism-espresso-integration/pull/57/commits/da01c1aa456665ca1e18386fea524fe272ebb5cb).

---

@shenkeyao  I did another update [760dcb1](https://github.com/EspressoSystems/optimism-espresso-integration/pull/57/commits/760dcb1930201d8bd875bedb96b37e76efa9bf0a), when the file does not exist, it actually return err. So err is expected when running in enclave.